shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -461,19 +461,38 @@ - (void)dispatchMouseEvent:(NSEvent*)event phase:(FlutterPointerPhase)phase {
   }
 }
 
-- (void)dispatchKeyEvent:(NSEvent*)event ofType:(NSString*)type {
+- (void)dispatchKeyEvent:(NSEvent*)event {
+  if (event.type != NSEventTypeKeyDown && event.type != NSEventTypeKeyUp) {
+    return;
+  }
   NSMutableDictionary* keyMessage = [@{
     @"keymap" : @"macos",
-    @"type" : type,
+    @"type" : event.type == NSEventTypeKeyDown ? @"keydown" : @"keyup",
     @"keyCode" : @(event.keyCode),
     @"modifiers" : @(event.modifierFlags),
   } mutableCopy];
-  // Calling these methods on any other type of event will raise an exception.
-  if (event.type == NSEventTypeKeyDown || event.type == NSEventTypeKeyUp) {
-    keyMessage[@"characters"] = event.characters;
-    keyMessage[@"charactersIgnoringModifiers"] = event.charactersIgnoringModifiers;
-  }
-  [_keyEventChannel sendMessage:keyMessage];
+  keyMessage[@"characters"] = event.characters;
+  keyMessage[@"charactersIgnoringModifiers"] = event.charactersIgnoringModifiers;
+  FlutterReply replyHandler = ^(id _Nullable reply) {
+    if (reply == nil) {
+      return;
+    }
+    if (![[reply valueForKey:@"handled"] boolValue]) {
+      // Only re-dispatch the event to other responders if the framework didn't handle it.
+      for (NSResponder* responder in self.additionalKeyResponders) {
+        if (event.type == NSEventTypeKeyDown) {
+          if ([responder respondsToSelector:@selector(keyDown:)]) {
+            [responder keyDown:event];
```

Comment:
ObjC noob question: Can I use the event object like this?  Will it be properly captured/released?